### PR TITLE
Update explainer to reflect up-to-date state of the API as of M111

### PR DESCRIPTION
**Pull request by GrapeGreen**

### CODE REVIEW

File: docs/explainer.md [hunk 1]
```diff
@@ -361,8 +334,36 @@ dictionary UDPMessage {
...
+```javascript
+let udpSocket = new UDPSocket({ remoteAddress, remotePort });
+let { readable } = await udpSocket.opened;
+
+let reader = readable.getReader();
+
+let { value, done } = await reader.read();
+if (done) {
+  // stream is exhausted...
+  // happens either on socket errors or explicit socket.close();
```
Comment (reillyeon) [inline docs/explainer.md:348]: UDP sockets and TCP server sockets can never be exhausted and `socket.close()` cannot be called while the reader is locked, which effectively means that `read()` will never return with `done == true`. The only way these socket types can be closed while they are being read from is if there is a network error, which means `read()` throws an exception.

Only TCP sockets can be exhausted when the remote peer closes the connection.

File: docs/explainer.md [hunk 2]
```diff
@@ -372,12 +373,14 @@ if (done) {
...
+}
 
 const decoder = new TextDecoder();
 
-// value is a UDPMessage object.
-let { data, remoteAddress, remotePort } = value;
+// |value| is a UDPMessage object.
+// |remoteAddress| and |remotePort| members of UDPMessage indicate the remote host
+// where the datagram came from.
+let { data, remoteAddress, remoetPort } = value;
```
Comment (reillyeon) [inline docs/explainer.md:383]:
```suggestion
let { data, remoteAddress, remotePort } = value;
```

File: docs/explainer.md [hunk 3]
```diff
@@ -388,27 +391,159 @@ let message = decoder.decode(data);
...
+
+const encoder = new TextEncoder();
+let message = "Some user-created datagram";
+
+// Sends a UDPMessage object where |data| is a Uint8Array.
+// Note that both |remoteAddress| and |remotePort| must be specified in this case.
+await writer.ready;
+writer.write({
+    data: encoder.encode(message),
+    remoteAddress: 'example.com',
```
Comment (reillyeon) [inline docs/explainer.md:439]: Add a note that specifying a domain name as the `remoteAddress` for an unconnected socket is inefficient as it requires a DNS lookup for each packet. Applications should try to replace the domain with the IP address of the peer after receiving a response packet.

File: docs/explainer.md [hunk 4]
```diff
@@ -388,27 +391,159 @@ let message = decoder.decode(data);
...
+let tcpServerSocket = new TCPServerSocket('::');
+let { readable: tcpServerSocketReadable } = await udpSocket.opened;
+
+let tcpServerSocketReader = tcpServerSocketReadable.getReader();
+
+// |value| is an accepted TCPSocket.
+let { value: tcpSocket, done } = await tcpServerSocketReader.read();
+if (done) {
+  // stream is exhausted...
+  // happens either on socket errors or explicit socket.close();
```
Comment (reillyeon) [inline docs/explainer.md:531]: Same here.

**Review by reillyeon — APPROVED** [review]

**Comment by Bamers25** [comment]
I have identity theft and I have to show my credentials to prove that I am a American citizen.  I'm a Army veteran.  This is gone to far

**Comment by Bamers25** [comment]
And I don't have Children.  . This is insane.

**Pull request merged**